lib/widgets/edit_state_marker.dart

```diff
@@ -0,0 +1,127 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter_gen/gen_l10n/zulip_localizations.dart';
+
+import '../api/model/model.dart';
+import 'icons.dart';
+import 'theme.dart';
+
+class SwipableMessageRow extends StatefulWidget {
+  const SwipableMessageRow({
+    super.key,
+    required this.child,
+    required this.message,
+  });
+
+  final Widget child;
+  final Message message;
+
+  @override
+  State<StatefulWidget> createState() => _SwipableMessageRowState();
+}
+
+class _SwipableMessageRowState extends State<SwipableMessageRow> {
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context).extension<DesignVariables>()!;
+
+    // TODO(#157): fix how star marker aligns with message content
+    // Design from Figma at:
+    //   https://www.figma.com/file/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=813%3A28817&mode=dev .
+    var star = Padding(padding: const EdgeInsets.only(top: 5.5),
+      child: Icon(ZulipIcons.star_filled, size: 16, color: theme.starColor));
+    final hasMarker = widget.message.editState != MessageEditState.none;
```

Comment:
Hmm, this file is called edit_state_marker.dart, so I'm wondering if the message-star UI code quite belongs in it. Maybe we want a different name for the file? @gnprice might have thoughts on the organization here.

Reply:
Yeah "edit_state_marker" is not a great fit. We could name it as with related to `SwipableMessageRow`, which is the only widget supposed to be exposed.

Reply:
How about having the caller stay responsible for the star? Then the child it passes here can include the star as well as the message content and reactions.